includes/Modules/Idea_Hub.php

```diff
@@ -277,6 +316,50 @@ protected function create_data_request( Data_Request $data ) {
 		return parent::create_data_request( $data );
 	}
 
+	/**
+	 * Parses a response for the given datapoint.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Data_Request $data     Data request object.
+	 * @param mixed        $response Request response.
+	 *
+	 * @return mixed Parsed response data on success, or WP_Error on failure.
+	 */
+	protected function parse_data_response( Data_Request $data, $response ) {
+		switch ( "{$data->method}:{$data->datapoint}" ) {
+			case 'GET:draft-post-ideas':
+				return array_map(
+					function( $post_id ) {
+						return array_merge(
+							array(
+								'postID'      => $post_id,
+								'postEditURL' => get_edit_post_link( $post_id ),
+							),
+							$this->get_post_idea( $post_id )
+						);
+					},
+					$response
+				);
+			case 'GET:published-post-ideas':
+				return array_map(
+					function( $post_id ) {
+						return array_merge(
+							array(
+								'postID'      => $post_id,
+								'postEditURL' => get_edit_post_link( $post_id ),
+								'postURL'     => get_permalink( $post_id ),
+							),
+							$this->get_post_idea( $post_id )
+						);
+					},
+					$response
```

Comment:
Same here, let's make sure it is an array:
```suggestion
					is_array( $response ) ? $response : array( $response )
```

Reply:
Updated @eugene-manuilov Can you check?